### PR TITLE
Google SignIn, User Creation, Basic page routing

### PR DESCRIPTION
**Pull request by talyh**
Google SignIn button connected to TOJam Firebase
Creates user record if it's a new user, with name, email and jamsAttended with the currentYear
Registration page is protected and requires user to be signed in
SignIn redirects to Registration page
Registration page shows user name as received from Google Auth

**Next Steps**
Improve state sharing with redux
Build form on registration page
Allow sign out

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,16 +3,29 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
+    "@fortawesome/fontawesome-svg-core": "^1.2.26",
+    "@fortawesome/free-brands-svg-icons": "^5.12.0",
+    "@fortawesome/free-solid-svg-icons": "^5.12.0",
+    "@fortawesome/react-fontawesome": "^0.1.8",
```
Comment (rota-rossi) [inline package.json]: Consider using `react-native-vector-icons`... it's really good. :)

File: package.json [hunk 2]
```diff
@@ -3,16 +3,29 @@
...
     "@testing-library/jest-dom": "^4.2.4",
     "@testing-library/react": "^9.3.2",
     "@testing-library/user-event": "^7.1.2",
     "@types/jest": "^24.0.0",
     "@types/node": "^12.0.0",
     "@types/react": "^16.9.0",
     "@types/react-dom": "^16.9.0",
+    "@types/react-redux": "^7.1.7",
+    "@types/react-router-dom": "^5.1.3",
+    "@types/styled-components": "^4.4.2",
```
Comment (rota-rossi) [inline package.json]: the types should be on `devDependencies`, as they are not needed in the build process.

File: src/components/buttons/SignInButton.tsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+  const result: any = await firebase.auth().signInWithPopup(provider);
+
+  // save token so we can skip login if jwt hasn't expired yet
+  localStorage.setItem("userToken", result.credential.idToken);
+  return result?.user;
+};
+
+const checkIfNewUser = async (uid: string) => {
+  // attempt to retrieve user record to determine if they've created a profile before
+  // TODO - Find a way to abstract this so it's not repeated throughout
```
Comment (rota-rossi) [inline src/components/buttons/SignInButton.tsx]: maybe migrate this to a `utils` file, and import this method from there.

File: src/serviceWorker.js [hunk 4]
```diff
@@ -20,18 +20,10 @@ const isLocalhost = Boolean(
     )
```
Comment (rota-rossi) [inline src/serviceWorker.js]: if you will not be using WebWorkers, you can safely remove this file.

File: package.json (hunk 1 above)
Comment (talyh) [reply package.json]: Nope. Not an RN project. 😄

File: src/serviceWorker.js (hunk 4 above)
Comment (talyh) [reply src/serviceWorker.js]: 👍

File: src/components/buttons/SignInButton.tsx (hunk 3 above)
Comment (talyh) [reply src/components/buttons/SignInButton.tsx]: Will do if the whole function needs abstraction.
For now, just the collection reference needs abstracting, but depending on where it goes, it conflicts with `firebase.initializeApp`, so haven't focused on finding the right place for it yet.

File: package.json (hunk 2 above)
Comment (talyh) [reply package.json]: 👍

**Pull request merged**